ksql-api/src/main/java/io/confluent/ksql/api/server/ReactiveSubscriber.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.server;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import java.util.Objects;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A reactive streams subscriber which handles much of the plumbing for you. Override {@link
+ * #afterSubscribe}, {@link #handleValue}, {@link #handleComplete} and {@link #handleError} to
+ * create your specific implementation.
+ *
+ * @param <T> The type of the value
+ */
+public class ReactiveSubscriber<T> implements Subscriber<T> {
```

Comment:
Ack, I'll add some extra javadoc

Reply:
Fwiw there's already some javadoc on the constructor which explains a bit, but I'll add some more to the header.
Please note, the implementation is in an early stage of development and is likely to change before production. Usually, I wouldn't add any JavaDoc until things have stabilised as otherwise you end up rewriting it every 5 minutes (and worse, forgetting to rewrite it resulting in out of date javadoc!), but I've made an exception here as I know most folks aren't too familiar with Vert.x and/or reactive streams so this helps the learning process :)